servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -342,104 +368,208 @@ public void registerBlockStoreEventListener(BlockStoreEventListener listener) {
   }
 
   /**
-   * Validate blockId is a temporary block and owned by userId, if the validation succeeds, return
-   * the {@link tachyon.worker.block.meta.TempBlockMeta} of blockId.
+   * Check if a blockId is available as a new temp block.
+   *
+   * @param blockId the ID of block
+   * @throws AlreadyExistsException if blockId already exists
+   */
+  private void checkTempBlockIdAvailable(long blockId) throws AlreadyExistsException {
+    mMetadataLock.readLock().lock();
+    try {
+      if (mMetaManager.hasTempBlockMeta(blockId)) {
+        throw new AlreadyExistsException("TempBlockMeta blockId " + blockId + " exists");
+      }
+      if (mMetaManager.hasBlockMeta(blockId)) {
+        throw new AlreadyExistsException("TempBlockMeta blockId " + blockId + " committed");
+      }
+    } finally {
+      mMetadataLock.readLock().unlock();
+    }
+  }
+
+  /**
+   * Check if blockId is a temporary block and owned by userId, return the
+   * {@link tachyon.worker.block.meta.TempBlockMeta} of blockId if the validation succeeds.
    *
    * @param userId the ID of user
    * @param blockId the ID of block
+   * @return the {@link tachyon.worker.block.meta.TempBlockMeta} of blockId
    * @throws NotFoundException if blockId can not be found in temporary blocks
    * @throws AlreadyExistsException if blockId already exists in committed blocks
    * @throws InvalidStateException if blockId is not owned by userId
-   * @return the {@link tachyon.worker.block.meta.TempBlockMeta} of blockId
    */
-  private TempBlockMeta validateUserTempBlock(long userId, long blockId) throws NotFoundException,
+  private TempBlockMeta checkTempBlockExists(long userId, long blockId) throws NotFoundException,
```

Comment:
This seems more like a "getTempBlock" or "getUserTempBlock" or something. What do you think?